classes/external_api.php

```diff
@@ -0,0 +1,360 @@
+ *
+ * @package     block_coursefeedback
+ * @copyright   2022 onwards Felix Di Lenarda (@ innoCampus, TU Berlin)
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace block_coursefeedback;
+
+defined('MOODLE_INTERNAL') || die();
+
+require_once("$CFG->libdir/externallib.php");
+require_once("$CFG->dirroot/webservice/externallib.php");
+require_once(__DIR__ . "/../lib.php");
+
+use external_value;
+use external_single_structure;
+use external_multiple_structure;
+use context_course;
+use context_system;
+use stdClass;
+
+/**
+ * Block coursefeedback external functions
+ *
+ * @package     block_coursefeedback
+ * @copyright   2022 onwards Felix Di Lenarda (@ innoCampus, TU Berlin)
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+// TODO  self::validate_parameters() externallib Zeile 324
+class external_api extends \external_api {
+
+    /**
+     * Returns description of answer_question_and_get_new parameters.
+     *
+     * @return external_function_parameters
+     */
+    public static function answer_question_and_get_new_parameters() {
+        $courseid = new external_value(
+            PARAM_INT,
+            'Courseid',
+            VALUE_REQUIRED
+        );
+        $feedback = new external_value(
+            PARAM_INT,
+            'Feedback',
+            VALUE_REQUIRED
+        );
+        $feedbackid = new external_value(
+            PARAM_INT,
+            'FeedbackID',
+            VALUE_REQUIRED
+        );
+        $questionid = new external_value(
+            PARAM_INT,
+            'QuestionID',
+            VALUE_REQUIRED
+        );
+        $params = array(
+            'courseid' => $courseid,
+            'feedback' => $feedback,
+            'feedbackid' => $feedbackid,
+            'questionid' => $questionid
+        );
+
+        return new \external_function_parameters($params);
+    }
+
+    /**
+     * Saves Feedback answer and returns the next feedback question if there is one.
+     *
+     * @param int $courseid
+     * @param int $feedback given answer
+     * @param int $feedbackid
+     * @param int $questionid
+     * @returns array The next questiondetails
+     */
+    public static function answer_question_and_get_new($courseid, $feedback, $feedbackid,  $questionid) {
+        global $DB, $USER;
+        
+        // Validate parameter
+        $params = self::validate_parameters(self::answer_question_and_get_new_parameters(),
+            array(
+                'courseid' => $courseid,
+                'feedback' => $feedback,
+                'feedbackid' => $feedbackid,
+                'questionid' => $questionid
+            )
+        );
+
+        // Security checks
+        $context = context_course::instance($params['courseid']);
+        self::validate_context($context);
+        require_capability('block/coursefeedback:evaluate', $context);
+
+        $result = array(
+            'saved' => false
+        );
+
```

Comment:
Bitte überprüfe, ob das Feedback überhaupt existiert, ob es zu dem angegebenen Kurs gehört, ob es zur Zeit aktiv ist, und ob bereits eine Antwort von dieser Person existiert, bevor du die Antwort abspeicherst

Reply:
Ergänzt in: d736c360fceb31dd7d894cedb97cb8a643296508

Reply:
Du musst noch überprüfen, ob das Feedback zu dem angegebenen Kurs gehört. Sonst kann man in beliebigen Kursen das Feedback beantworten, indem man die ID eines anderen Kurses übergibt, in den man eingeschrieben ist.

Reply:
Ergänzt in: a4ae28364fb16b266bd08e518595f3c4c8181608

Reply:
Aber jetzt hast du immer noch nicht überprüft, ob das _Feedback_ zu dem Kurs gehört. `$COURSE` enthält nur den aktuellen Kurs, in dem sich der User befindet.